### PR TITLE
[refactor]: iroha_crypto dependencies

### PR DESCRIPTION
**Pull request by mversic**
### Description of the Change

* remove `serde_json` as dependency of `iroha_crypto`
* remove `openssl-sys` as direct dependency of `iroha_crypto`
* don't implement `TryFrom` for references to `PublicKey` and `Multihash`, use owned types

### CODE REVIEW

**Comment by mversic** [comment]
It builds for me locally, but not in CI. Should be looked into

File: crypto/src/lib.rs [hunk 1]
```diff
@@ -39,7 +39,7 @@ pub const BLS_NORMAL: &str = "bls_normal";
 pub const BLS_SMALL: &str = "bls_small";
 
 /// Algorithm for hashing
-#[derive(Clone, Copy, Debug)]
+#[derive(Debug, Clone, Copy)]
```
Comment (appetrosyan) [inline crypto/src/lib.rs:42]: Anything wrong with the alphabetical order? I'm just curious.

Comment (appetrosyan) [reply crypto/src/lib.rs:42]: Perhaps we could add a minor point to the style guide if having `Debug` before `Clone` and `Copy` has an even minor impact.

Comment (mversic) [reply crypto/src/lib.rs:42]: ahm, not that it really matters, I just tried enforcing some consistency in derive order, my rules were:
1. most frequently used items first
2. if derive depends on another derive that one goes first, like `Clone, Copy` or `PartialEq, Eq`

**Review by appetrosyan — APPROVED** [review]

**Review by s8sato — APPROVED** [review]

**Pull request merged**